geozero/src/wkt/mod.rs

```diff
@@ -9,25 +9,46 @@ pub use wkt_writer::*;
 
 pub(crate) mod conversion {
     use crate::error::Result;
-    use crate::wkt::WktWriter;
+    use crate::wkt::{WktDialect, WktWriter};
     use crate::{CoordDimensions, GeozeroGeometry};
 
     /// Convert to WKT.
     pub trait ToWkt {
         /// Convert to 2D WKT String.
         fn to_wkt(&self) -> Result<String>;
+        /// Convert to EWKT String.
+        fn to_ewkt(&self, srid: Option<i32>) -> Result<String>;
         /// Convert to WKT String with dimensions.
         fn to_wkt_ndim(&self, dims: CoordDimensions) -> Result<String>;
+        /// Convert to WKT String with srid, dimensions and dialect.
+        fn to_wkt_dialect(
```

Comment:
Technically this is more of a `to_wkt_with_options` or `to_wkt_extended` method, not just `dialect`... Naming is hard, we may want to bikeshed this one.... (note that I am still kinda uneasy about the `new` constructor now requiring an additional dialect param even though most of the users probably won't ever care about it, and plus it doesn't allow us to have a default

Reply:
I just hope the number of dialects don't explode :)  `to_ewkt` and `to_ewkt_ndim` sounds fine.

Reply:
sigh :)  I think we may as well just use `to_wkt_with_opts`, and provide all those detalis. We may even want to get rid of the `to_wkt_ndims` to reduce the api surface (TBD)

I just really don't like the `to_wkt_dialect` because it actually has many more params than just the dialect one, making it confusing

Reply:
I agree, not sure what the best solution is here so I'll that one up to you..!